mitmproxy/proxy/layers/http/__init__.py

```diff
@@ -351,7 +351,7 @@ def check_killed(self, emit_error_hook: bool) -> layer.CommandGenerator[bool]:
                 yield HttpErrorHook(self.flow)
             # For HTTP/2 we only want to kill the specific stream, for HTTP/1 we want to kill the connection
             # *without* sending an HTTP response (that could be achieved by the user by setting flow.response).
-            if self.context.client.alpn == b"h2":
+            if self.context.client.alpn == "h2":
```

Comment:
See above comment: technically this should be a byte string. There is no guarantee that bytes->utf8->bytes conversions will match again. Not only replace-encoding, but also different utf-8 code points etc.